rust/xaynet-server/src/storage/redis/mod.rs

```diff
@@ -70,115 +77,100 @@ use xaynet_core::{
     UpdateParticipantPublicKey,
 };
 
+/// Redis client.
 #[derive(Clone)]
 pub struct Client {
-    raw_connection: ConnectionManager,
-    semaphore: Arc<Semaphore>,
+    connection: ConnectionManager,
 }
 
 #[cfg(test)]
 impl std::fmt::Debug for Client {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.debug_struct("Client")
-            .field("semaphore", &self.semaphore)
-            .finish()
+        f.debug_struct("redis client").finish()
     }
 }
 
-pub struct Connection {
-    connection: ConnectionManager,
-    _permit: OwnedSemaphorePermit,
+fn to_storage_err(e: RedisError) -> StorageError {
```

Comment:
implementing `From<RedisError>` for `StorageError` feels a bit more natural imo

Reply:
At first I implemented that way but then I noticed I can't because of the orphan rule

Reply:
ahh because `StorageError` is an alias for `anyhow::Error`. Fair enough then.